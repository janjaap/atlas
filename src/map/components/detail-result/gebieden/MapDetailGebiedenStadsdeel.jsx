import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultWrapper from '../MapDetailResultWrapper';

const MapDetailGebiedenStadsdeel = ({ panoUrl, stadsdeel, onMaximize, onPanoPreviewClick }) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={stadsdeel.label}
    title="Stadsdeel"
  />
);

MapDetailGebiedenStadsdeel.propTypes = {
  stadsdeel: PropTypes.shape({
    label: PropTypes.string.isRequired
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailGebiedenStadsdeel;
