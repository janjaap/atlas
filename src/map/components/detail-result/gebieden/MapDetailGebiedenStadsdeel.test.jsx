import React from 'react';
import { shallow } from 'enzyme';

import MapDetailGebiedenStadsdeel from './MapDetailGebiedenStadsdeel';

describe('MapDetailGebiedenStadsdeel', () => {
  it('should render everything', () => {
    const stadsdeel = {
      label: 'Stadsdeel label'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailGebiedenStadsdeel
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        stadsdeel={stadsdeel}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
