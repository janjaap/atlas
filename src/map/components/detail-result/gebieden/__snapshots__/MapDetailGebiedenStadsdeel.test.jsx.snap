// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailGebiedenStadsdeel should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Stadsdeel label"
  title="Stadsdeel"
/>
`;
